File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class RnaTranscription
{
    private static Dictionary<char, char> _dnaToRna = new Dictionary<char, char> {
        {'G', 'C'}, {'C', 'G'}, {'T', 'A'}, {'A', 'U'}
    };
    public static string ToRna(string strand)
    {
        return string.Join("",strand.Select(n => DnaToRna[n]));
    }
}

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string): string =
    let mapping ch =
        match ch with
        | 'G' -> 'C'
        | 'C' -> 'G'
        | 'T' -> 'A'
        | 'A' -> 'U'
        | _ -> '_'
    dna |> String.map mapping